code/controllers/subsystem/economy.dm

```diff
@@ -15,121 +17,109 @@ SUBSYSTEM_DEF(economy)
+	var/centcomm_fund = base_station_income
+	centcomm_fund += maintenance_bonus()  // Payout based on integrity.
+	centcomm_fund += research_bonus() // Payout based on slimes.
+	centcomm_fund += crew_status_bonus() // Payout based on crew safety, health, and mood.
+	centcomm_fund += next_income_bonus
+	last_income = centcomm_fund
+	next_income_bonus = 0
+	
+	distribute_income()
 	for(var/A in bank_accounts)
 		var/datum/bank_account/B = A
 		B.payday(1)
 
+/datum/controller/subsystem/economy/proc/distribute_income(income)
+	station_budget.adjust_money(income * income_distribution[station_budget.department_id])
+	for(var/X in generated_accounts)
+		var/datum/bank_account/department/B = X
+		B.adjust_money(income * income_distribution[B.department_id])	
 
+//Change a department's income distribution. The remaining income is given to the station's account.
+/datum/controller/subsystem/economy/proc/change_distribution(datum/bank_account/department/D, new_value)
+	var/set_value = CLAMP(ROUND(new_value), 0, 100)
+	var/current_value = income_distribution[D.department_id]
+	if(D == station_budget) //This is what's left, and isn't modified directly
+		return FALSE
+	if(set_value == current_value) //that was easy
+		return TRUE
+	else if(set_value < current_value)
+		var/difference = current_value - set_value
+		income_distribution[D.department_id] -= difference
+		income_distribution[station_budget.department_id] += difference
+		return TRUE
+	else
+		var/difference = set_value - current_value
+		if(income_distribution[station_budget.department_id] >= difference)
+			income_distribution[D.department_id] += difference
+			income_distribution[station_budget.department_id] -= difference
+			return TRUE
+		else
+			return FALSE //out of budget
+		
 /datum/controller/subsystem/economy/proc/get_dep_account(dep_id)
 	for(var/datum/bank_account/department/D in generated_accounts)
 		if(D.department_id == dep_id)
 			return D
 
-/datum/controller/subsystem/economy/proc/boring_eng_payout()
+/datum/controller/subsystem/economy/proc/maintenance_bonus()
 	var/engineering_cash = 3000
 	engineering_check.count()
 	var/station_integrity = min(PERCENT(GLOB.start_state.score(engineering_check)), 100)
 	station_integrity *= 0.01
 	engineering_cash *= station_integrity
-	var/datum/bank_account/D = get_dep_account(ACCOUNT_ENG)
-	if(D)
-		D.adjust_money(engineering_cash)
+	return engineering_cash
 
-/datum/controller/subsystem/economy/proc/boring_secmedsrv_payout()
-	var/crew
-	var/alive_crew
-	var/dead_monsters
-	var/cash_to_grant
-	for(var/mob/m in GLOB.mob_list)
-		if(isnewplayer(m))
-			continue
-		if(m.mind)
-			if(isbrain(m) || iscameramob(m))
-				continue
-			if(ishuman(m))
-				var/mob/living/carbon/human/H = m
-				crew++
-				if(H.stat != DEAD)
-					alive_crew++
-					GET_COMPONENT_FROM(mood, /datum/component/mood, H)
-					var/medical_cash = (H.health / H.maxHealth) * alive_humans_bounty
-					if(mood)
-						var/datum/bank_account/D = get_dep_account(ACCOUNT_SRV)
```

Comment:
I'm not seeing where mood bounties are even being paid out here, or medical bounties.

Reply:
You need to re-add these. Otherwise you've now killed any incentive for Medical or Service to do their jobs.

Reply:
To be honest i doubt that overall mood is very influenced by service jobs, as long as most people won't partake in their product. Which they generally won't if it has no mechanical benefit. I think there might be a more direct solution, but i don't have a concrete replacement yet

Medical bounty is mostly in keeping people alive and on sensors. Since people are incentivized in getting sensors, medical will have more information to act on when trying to save people. People who play in medbay, in my experience, generally do it because they like healing people in the first place, so i believe this might be a better option.

Reply:
The only thing service jobs impact IS mood. Food, drugs, drinks, all 3 of them only impact mood. Same with cleaning the station.

And medical bounty just counts for being alive, and cloning, which is what medical just does already. Making them get more from healing incentivizes people who normally half-ass it to full-ass it.

Reply:
I still think that centcomm directly knowing people's moods is too inconsistent with their current portrayal. Plus, again, direct incentives are more effective than trying to improve a very wide statistic when you can only affect the subset of people who care about your service.

I'm still thinking about possible replacements for service following that reasoning.
Perhaps there could be a space instagram so you can share photos of your rare food/drinks to centcomm and get rewarded.

Reply:
Gameplay always takes priority over realism/roleplay here.

Reply:
dare I say it,

I agree with goof and realism is a stupid thing to limit something like this over.

Reply:
have a suit sensors check so you can fluff it as "they're reading brainwaves" or something.

Reply:
I have no idea why, but for some reason, a direct check on mood affecting the budget makes sense to me. I am probably the person *most* in favor of realism on the server, so this is kind of strange.

I still would probably prefer some other mechanic for it though, if possible.

Reply:
~they monitor the radio chatter for morale~